### DISCUSSION TITLE
🦇 BAT: Bayesian-estimate Adjustment Time in Nonlinear dynamic for decision

### DISCUSSION THREAD

**Discussion by hyunjimoon** [Data4DM/BayesSD#19] [perspectives; sense-making]
a. do you view point (9, 6) as chaos? 

b. could we define non-identifiability in dynamic model using BATS (e.g. \exists two different neighboring pattern classes) so that its existence can be iff condition for chaos?

c. how to set the segment/time horizon/aggregation level that enables this identifiability (p.46, 36 of [actionable workflow slides](https://github.com/Data4DM/BayesSD/blob/master/DailyDigest/ActionableWorkflow_Moon.pdf) are related and I believe this should be reverse engineered, unlike current convention of using default)

![image](https://user-images.githubusercontent.com/30194633/191912094-2285a964-716b-4d03-8d2c-c39fc391e232.png)

**Comment by tomfid** [comment]
I think it's difficult for a classifier to identify chaos from a single run. It might be possible to compute some kind of fingerprint (like a fractal dimension) but that would normally require a much longer time series than we typically have in an SD model of a real problem. It's a very interesting question to consider if we want to cover dynamic systems very generally.

**Reply by hyunjimoon** [reply]
Could comparing the parameter vector distance and stock tensor distance identify reinforcement loop?

If small change in parameter results in large difference in stock tensor (different reference mode), it needs attention!

**Reply by tomfid** [reply]
Doesn't eigenvalue elasticity do this more directly?

**Reply by hyunjimoon** [reply]
Yes but definition of distance (divergence) is VERY important. 
Comparison between different data structure would happen (as distance(1,7), distance(2,6), distance(3,5)) in examples below) and should be taken into care. For instance, distance between time series vector, correlation is usually used but I think 
1. decompose time series into trend, seasonality, event component to estimate each component's parameter
2. measure distance between parameters 

would be more precise. For one, it can avoid the shifted oscillatory reference modes returning low value of correlation.

<img width="707" alt="image" src="https://user-images.githubusercontent.com/30194633/194609547-59f986f1-ffb2-4225-898f-e8bb2201709a.png">

**Comment by hyunjimoon** [comment]
@tomfid Regarding our discussion on inferring structure from reference mode, Yaman's following mapping of `stock-flow`:`causal loop`: `reference mode` could give us some idea. (your blog on bathtub dynamics as well).

Major three dynamics: management of diffusion, inventory (time-chain), prey-predator (space-chain).
![image](https://user-images.githubusercontent.com/30194633/194714626-9f1c5ba5-3ebc-4d11-abe6-3f3a12396fab.png)

**Reply by hyunjimoon** [reply]
I wish to ask for some feedback on:
1. most decisions (policy levels) are in the form of adjustment time 
2. `adj time` can be equally represented as `frac` (in above fig, removal, contact, transition etc)

**Comment by hyunjimoon** [comment]
Below are relevant Gönenç's pattern recognition and behavior space exploration. 

Edali M, Yücel G.  2020.  Analysis of an individual-based influenza epidemic model using random forest metamodels and adaptive sequential sampling. Systems Research and Behavioral Science. :1–23.
Edali M., Baydoğan M.G., Yücel G..  2019.  Classification of generic system dynamics model outputs via supervised time series pattern discovery. Turkish Journal of Electrical Engineering and Computer Sciences. 27
Edali M, Yücel G.  2019.  Exploring the behavior space of agent-based simulation models using random forest metamodels and sequential sampling. Simulation Modelling Practice and Theory. 92:62-81.
Edali M, Yücel G.  2018.  Automated Analysis of Regularities Between Model Parameters and Output Using Support Vector Regression in Conjunction with Decision Trees. Journal of Artificial Societies and Social Simulation. 21:1.

**Comment by hyunjimoon** [comment]
Regarding David's "then prediction of where a boundary will be is only possible by calculating the full fractal" from https://statmodeling.stat.columbia.edu/2013/02/06/the-fractal-nature-of-scientific-revolutions/, instead of predicting global boundaries, algorithms that keep agents from crossing the boundary (e.g. **tolerance decrease rate proportional to perceived uncertainty increase rate**) and having many agents delineate the boundary might be a usable approximation? You may ask how we know the boundary; scaling law i.e. information pooling across different organisms / fields can give hints.
